### ISSUE TITLE
browser extension

### ISSUE THREAD

**Issue by iawia002** [iawia002/annie#80]
just one click, construct the parameters(with cookies), then open the terminal

**Comment by noelyahan** [comment]
In this case only problem I see is the opening the terminal part. Because the browser is running inside the secure sandbox it wont allow to communicate with os processes. But for Internet Explorer ActiveX object we can handle it.

**Comment by noelyahan** [comment]
Hi @iawia002 I did some further research and found out 2 protocols that can communicate with native applications.

1.  [NPAPI](https://developer.chrome.com/extensions/npapi) [**Deprecated**]
2. [Native Messaging API](http://developer.chrome.com/extensions/messaging.html#native-messaging)

The NPAPI was deprecated and, **Native messaging API** is working fine with current chrome browsers.
![chrome-native](https://user-images.githubusercontent.com/6106461/43678251-934292ce-982d-11e8-9d94-42f1d2a920cc.png)

I would like to give try on this feature by developing the annie chrome extension.

**Comment by iawia002** [comment]
Nice! Thanks for doing this. You can give it a try, when you make some progress, please let me know

**Comment by noelyahan** [comment]
I implemented a chrome extension under [feature-browser-ext branch](https://github.com/noelyahan/annie/tree/feature-browser-ext/addons/chrome) :) [Demo](https://youtu.be/Y5O4BfmMgB4). And now I'm working on improvements (Pause, Resume, Stop, Delete, Open). I used intermediate host because currently I didn't wanted to mess up with the main.go app. Later we can avoid this intermediate host app and integrate with the main app.

- Note: downloads will goes to default ~/Downloads folder

**Comment by iawia002** [comment]
The demo looks fine!

However, I am a little worried about it. To use this, I need to install the host first, which may be a bit troublesome for the user.

So I have a suggestion, for all users, we show the constructed download parameters (with cookies, one more question here, I don't know if Chrome allows reading cookies?) which can be easily copied, for the user who installed host, we will show them a download button, they can download the video directly if they want.

**Comment by noelyahan** [comment]
I agree, It's better to provide both solutions. yes chrome extension allows to use cookies.
As an example if a user want's to download a playlist do the extension needs

1. Provide only the clipboard command and params to run the annie manually
2. Provide a UI to tweak some parameters such as quality, selected videos

**Comment by iawia002** [comment]
The first version doesn't need to be too complicated, maybe just a clipboard and a download button.

And one more thing, choose the `quality` may be a little tricky, you won't know what quality provided unless you run `annie -i URL...` command in the terminal.
